ViewController.swift

```diff
@@ -6,14 +6,156 @@
+        }
+    }
+    
+    func imagePickerControllerDidCancel(_ picker: UIImagePickerController) {
+        dismiss(animated: true){ [weak self] in
+            guard let self = self else { return }
+            self.stopLoadingAnimation()
+        }
+    }
+    
+    private func getDocumentsDirectory() -> URL? {
+       FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first
+    }
+    
+    override func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        	
+        let alertController = UIAlertController(title: nil, message: "Please Select an Option", preferredStyle: .actionSheet)
+        
+        alertController.addAction(UIAlertAction(title: "Rename", style: .default, handler: { [weak self] _ in
+            guard let self = self else { return }
+            let renameAlertViewController = self.makeRenameAlertViewController(indexPath: indexPath)
+            self.present(renameAlertViewController, animated: true)
+        }))
+        
+        alertController.addAction(UIAlertAction(title: "Delete", style: .destructive, handler: { [weak self, weak alertController] _ in
+            guard let self = self else { return }
+            self.people.remove(at: indexPath.item)
+            alertController?.dismiss(animated: true)
+            self.collectionView.reloadData()
+        }))
+        
+        alertController.addAction(UIAlertAction(title: "Cancel", style: .default, handler: { [weak alertController] _ in
+            alertController?.dismiss(animated: true)
+        }))
+    
+        present(alertController, animated: true)
+        
+    }
+    func makeRenameAlertViewController(indexPath: IndexPath) -> UIAlertController {
+        
+        let renameAlertController = UIAlertController(title: "Rename", message: "What's the new name?", preferredStyle: .alert)
+        renameAlertController.addTextField()
+        
+        renameAlertController.addAction(UIAlertAction(title: "Rename", style: .default) { [weak self, weak renameAlertController] _ in
+            guard let self = self, let newName = renameAlertController?.textFields?[0].text else { return }
+            self.people[indexPath.item].name = newName
+           
+            self.collectionView.reloadData()
+        })
+        
+        renameAlertController.addAction(UIAlertAction(title: "Cancel", style: .cancel, handler: { [weak renameAlertController] _ in
+            
+            renameAlertController?.dismiss(animated: true)
+          
+        }))
+        return renameAlertController
+    }
+    
+    private func startLoadingAnimation() { 
+        view.backgroundColor = UIColor.black.withAlphaComponent(0.5)
+        blurEffectView.translatesAutoresizingMaskIntoConstraints = false // This is an explicit declaration that auto resizing masks won't be used
+              loadingActivityIndicator.translatesAutoresizingMaskIntoConstraints = false
+              view.addSubview(blurEffectView)
```

Comment:
Oh, I forgot the nav bar in my suggestion, so the + button is still visible and available to be tapped. It's ok, but in case you wanted the blur view to cover the entire screen, you'd have to add it to the active window. You can see how to that here https://stackoverflow.com/a/37918334/4278574, but really, it's not worth it here.